plasmapy/particles/ionization_state.py

```diff
@@ -2,32 +2,168 @@
 Objects for storing ionization state data for a single element or for
 a single ionization level.
 """
-__all__ = ["IonizationState", "State"]
 
-import astropy.units as u
-import collections
+__all__ = ["IonizationState", "IonicFraction"]
+
 import numpy as np
 import warnings
 
+from astropy import units as u
 from numbers import Integral, Real
 from typing import List, Optional, Union
 
 from plasmapy.particles.decorators import particle_input
-from plasmapy.particles.exceptions import AtomicError, ChargeError, InvalidParticleError
+from plasmapy.particles.exceptions import (
+    ChargeError,
+    InvalidParticleError,
+    ParticleError,
+)
 from plasmapy.particles.particle_class import Particle
 from plasmapy.utils.decorators import validate_quantities
 
 _number_density_errmsg = (
     "Number densities must be Quantity objects with units of inverse " "volume."
 )
 
-# TODO: Change `State` into a class with validations for all of the
-# TODO: attributes.
 
-#: Named tuple class for representing an ionization state (`collections.namedtuple`).
-State = collections.namedtuple(
-    "State", ["integer_charge", "ionic_fraction", "ionic_symbol", "number_density"]
-)
+class IonicFraction:
+    """
+    Representation of the ionic fraction for a single ion.
+
+    This class is created when indexing
```

Comment:
This is what is returned when using `__getitem__` for an `IonizationState` instance.  I removed the line since that sort of explanation goes better in `IonizationState`'s docstring.